feat :  API 클라이언트 Mocking 하는 컴포넌트 추가 공통기능

shindh12 opened the pull request:
## What is this PR? 🎯
공통기능 추가를 위한 shared-library 프로젝트 세팅
API 클라이언트 Mocking 하는 컴포넌트 추가 공통기능

## Changes 📄
- library 네이밍 수정
- 각 gradle.build 에 필요한 디펜던시 및 설정 추가
- MockClient 어노테이션 추가
- 가이드 테스트 코드 추가

## Dependency 🔁


## Related Issues 👀


## Test Checklist ✔
- mock-context : MockClientTest 참조.

shindh12 committed:
[#34] 공통기능 추가를 위한 shared-library 추가 사전 작업

## What is this PR? 🎯
- 공통 기능 추가 전 프로젝트 세팅을 위한 gradle 세팅 추가 및 변경

## Changes 📄
- library 네이밍 수정
- 각 gradle.build 에 필요한 디펜던시 및 설정 추가

## Dependency 🔁

## Related Issues 👀

## Test Checklist ✔

shindh12 committed:
[#34] API 클라이언트 Mocking 하는 컴포넌트 추가 공통기능

## What is this PR? 🎯
- API 클라이언트 Mocking 하는 컴포넌트 추가 공통기능

## Changes 📄
- MockClient 어노테이션 추가
- 가이드 테스트 코드 추가

## Dependency 🔁

## Related Issues 👀

## Test Checklist ✔
- mock-context : MockClientTest 참조.

biiin commented on shared-library/stos-context/src/test/java/com/stos/concert/shared/support/mock/MockFoo.java:
```diff
@@ -0,0 +1,12 @@
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@MockClient
```
이런 방법이..!!  👀 👍👍👍

biiin commented:
2번째 commit은 별도 이슈로 작업해도 좋았을듯 !
[공통기능 추가를 위한 shared-library 추가 사전 작업](https://github.com/STOS-INC/concert-reservation/pull/36/commits/1493f6d1182b1e57139681093ea1d34b8233773b)

shindh12 committed:
[#37] hall service 프로젝트 추가 및 멀티프로젝트 구조 개선

## What is this PR? 🎯
- 멀티 프로젝트 빌드 제대로 안되는 구조라서 개선 필요

## Changes 📄
- gradle wrapper 루트 프로젝트로 사용하도록 수정

## Dependency 🔁

## Related Issues 👀
[#34] 이후 작업

## Test Checklist ✔

shindh12 committed:
[#37] hall service 프로젝트 추가 및 멀티프로젝트 구조 개선

## What is this PR? 🎯
- hall service 멀티 프로젝트 추가

## Changes 📄
- hall service module 추가됨

## Dependency 🔁

## Related Issues 👀
[#34] 이후 작업

## Test Checklist ✔

shindh12 committed:
[#33] 로컬 개발 환경 추가

## What is this PR? 🎯
- docker-compose에 인프라 개발 환경 추가

## Changes 📄
- compose.yaml 에 인프라 환경 세팅 (mysql, redis, kafka)
- 로컬에 생기는 data 디렉토리 gitignore 추가

## Dependency 🔁
[#37] 이후 작업

## Related Issues 👀
[#33]

## Test Checklist ✔
mysql : curl -v localhost:3306
kafka : curl -v localhost:9092
redis : curl -v localhost:6379
redis-stack : http://localhost:8001 브라우저 접속
kafka-ui : http://localhost:8081 브라우저 접속

shindh12 committed:
[#9] 예외 처리 공통로직 개발

## What is this PR? 🎯
- 예외 처리를 위한 공통로직 개발 > Exception 모델 추가

## Changes 📄
- Exception 모델 추가

## Dependency 🔁
[#33] 이후 작업

## Related Issues 👀
[#9]

## Test Checklist ✔

shindh12 committed:
[#9] autoconfigure 프로젝트 추가

## What is this PR? 🎯
- 예외 처리를 위한 공통로직 개발 > autoconfigure 프로젝트 추가

## Changes 📄
- autoconfigure 프로젝트 관련 세팅 추가 및 수정

## Dependency 🔁
[#33] 이후 작업

## Related Issues 👀
[#9]

## Test Checklist ✔

shindh12 committed:
[#9] 에러 및 exception 모듈 추가

## What is this PR? 🎯
- 예외 처리를 위한 공통로직 개발

## Changes 📄
- ControllerAdvice -> `GlobalExceptionHandler` 추가
- Error 형태에 따라 Resolve 해주는 `ErrorResolver` 모듈 추가
- Error Property 확장성을 위한 `ErrorProcessor` 인터페이스 추가
- AutoConfiguration 추가

## Dependency 🔁
[#33] 이후 작업

## Related Issues 👀
- issue : [#9]
- 에러 응답 표준 : https://datatracker.ietf.org/doc/html/rfc7807
- spring autoconfigure : https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#features.developing-auto-configuration

## Test Checklist ✔

shindh12 committed:
[#9] 에러 및 exception 모듈 추가

## What is this PR? 🎯
- 예외 처리를 위한 공통로직 개발 > README 추가

## Changes 📄
- 가이드를 위한 README 추가

## Dependency 🔁
[#33] 이후 작업

## Related Issues 👀
- issue : [#9]

## Test Checklist ✔

nicksh-woowahan committed:
[issue-17] 콘서트, 홀 관련 엔티티 설계

shindh12 committed:
[#9] 에러 및 exception 모듈 추가
## Changes 📄
- NotFound 에러 추가

shindh12 committed:
Merge pull request #42 from STOS-INC/feature/issue-9

feat: 예외 처리 공통로직 개발

shindh12 committed:
Merge pull request #39 from STOS-INC/feature/issue-33

feat: 로컬 개발 환경 추가

shindh12 committed:
Merge pull request #43 from STOS-INC/feature/issue-17

[issue-17] 콘서트, 홀 관련 엔티티 설계

shindh12 committed:
Merge pull request #38 from STOS-INC/feature/issue-37

feat : hall service 프로젝트 추가 및 멀티프로젝트 구조 개선
